api/routes/api/user.go

```diff
@@ -77,6 +77,70 @@ func CreateUserHandle(c *gin.Context) {
 	})
 }
 
+type createUserByAdminRequestBody struct {
+	Username string
+	Password string
+	Role     models.Role
+}
+
+func CreateUserByAdminHandle(c *gin.Context) {
+	var body createUserByAdminRequestBody
+
+	if err := c.BindJSON(&body); err != nil {
+		c.JSON(http.StatusOK, gin.H{
+			"message": "error",
+		})
+		return
+	}
+
+	newUser := models.User{
+		Id:   primitive.NewObjectID(),
+		Name: body.Username,
+		Role: body.Role,
+	}
+
+	err := models.CreateUser(&newUser)
+
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{
```

Comment:
.

Reply:
I'll check that. 

But I don't think we need authentication with any of these changes? Since they're only accessible via the admin tools